recipes/mosquitto/all/conanfile.py

```diff
@@ -0,0 +1,116 @@
+import os
+import glob
+from conans import CMake, ConanFile, tools
+
+class MosquittoConan(ConanFile):
+    name = "mosquitto"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/eclipse/mosquitto"
+    topics = ("mqtt", "broker", "libwebsockets", "mosquitto", "eclipse-iot")
+    license = "EPL-1.0"
+    description = "Eclipse Mosquitto - An open source MQTT broker"
+    exports_sources = ["CMakeLists.txt", "patches/*"]
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False],
+               "fPIC": [True, False],
+               "with_tls": [True, False]}
+    default_options = {"shared": False,
+                       "fPIC": True,
+                       "with_tls": True}
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.options.with_tls:
+            self.requires("openssl/1.1.1g")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name.replace("-", ".") + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+            self._cmake.definitions["WITH_STATIC_LIBRARIES"] = not self.options.shared
+            self._cmake.definitions["WITH_TLS"] = self.options.with_tls
+            self._cmake.definitions["DOCUMENTATION"] = False
+            self._cmake.definitions["WITH_THREADING"] = self.settings.compiler != "Visual Studio"
+            self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def _patch(self):
+        if not self.options.shared and not self.options.get_safe("fPIC"):
+            tools.replace_in_file(os.path.join(self._source_subfolder, "lib", "CMakeLists.txt"),
+                                  "POSITION_INDEPENDENT_CODE 1",
+                                  "POSITION_INDEPENDENT_CODE 0")
+        if self.settings.os == "Windows":
+            tools.replace_in_file(os.path.join(self._source_subfolder, "lib", "CMakeLists.txt"),
```

Comment:
this explains (and fixes) my problem in https://github.com/conan-io/conan-center-index/pull/3949
however, I think a patch should be created, and the change should be suggested upstream to the mosquitto project ,  hopefully I will find the time for that